src/pages/NewCodeView.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { Pressable, StyleSheet, Text, TextInput, View } from 'react-native';
+import { useFormik } from 'formik';
+import { NavigationProp } from '@react-navigation/core';
+import { CustomButton } from '../components/CustomButton';
+import { newCodeRequest } from '../services/user.services';
+import { useToastAlert } from '../hooks/useToastAlert';
+
+interface NewCodeViewProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  navigation: NavigationProp<any, any>;
+}
+
+export const NewCodeView = ({ navigation }: NewCodeViewProps) => {
+  const { showSuccessToast, showErrorToast } = useToastAlert();
+
+  const redirectToEmailVal = () => {
+    navigation.navigate('EmailValidation');
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      email: ''
+    },
+    // todo
+    onSubmit: async (values) => {
+      const [response, error] = await newCodeRequest(values.email);
+      if (error && response?.message) {
+        showErrorToast(response?.message);
+      } else {
+        showSuccessToast(response?.message);
+        redirectToEmailVal();
+      }
+    }
+  });
+  return (
+    <View style={Styles.container}>
+      <View style={Styles.header}>
+        <Text style={Styles.headerTitle}>INSERT YOUR EMAIL</Text>
```

Comment:
The main title of the view should be something like "Request a new code", so the user knows what the purpose of the view is.

Reply:
You are right, I fix it!